spec/slack-ruby-bot/hooks/message_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'spec_helper'
+
+class DummyBot < SlackRubyBot::Bot; end
+describe SlackRubyBot::Hooks::Message do
+  let(:server) { SlackRubyBot::Server.new }
+  describe '#child_command_classes' do
+    it 'returns only child command classes' do
+      child_command_classes = server.send(:child_command_classes)
+      expect(child_command_classes).to include(SlackRubyBot::Bot)
+      expect(child_command_classes).to include(DummyBot)
+      expect(child_command_classes).to_not include(SlackRubyBot::Commands::Hi)
+    end
+  end
+  describe '#built_in_command_classes' do
+    let(:built_in_command_classes) { server.send(:built_in_command_classes) }
+    it 'returns only built in command classes' do
+      expect(built_in_command_classes).to include(SlackRubyBot::Commands::Hi)
+      expect(built_in_command_classes).to include(SlackRubyBot::Commands::Default)
+      expect(built_in_command_classes).to include(SlackRubyBot::Commands::Help)
+      expect(built_in_command_classes).to_not include(SlackRubyBot::Bot)
+    end
+    it 'does not return unknown command class' do
+      expect(built_in_command_classes).to_not include(SlackRubyBot::Commands::Unknown)
+    end
+  end
+  describe '#message_to_self_not_allowed?' do
+    context 'if allow_message_loops is true' do
+      it do
+        SlackRubyBot::Config.allow_message_loops = true
+        expect(server.send(:message_to_self_not_allowed?)).to eq(false)
+      end
+    end
+    context 'if allow_message_loops is false' do
+      it do
+        SlackRubyBot::Config.allow_message_loops = false
```

Comment:
You don't want to set something globally and then not reset it because it will have side-effects on other specs. This should be done "in context":

```ruby
context 'with allow_message_loops set to false' do
  before do
    SlackRubyBot::Config.allow_message_loops = false
  end
  after do
    # at least: SlackRubyBot::Config.allow_message_loops = nil
    SlackRubyBot::Config.reset! # better, needs to be implemented
  end
```